test/e2e-shared-tests/mcd_config_drift.go

```diff
@@ -218,6 +231,62 @@ func (c configDriftTest) Run(t *testing.T) {
 				})
 			},
 		},
+		{
+			name:           "update degraded reason on drift",
+			rebootExpected: false,
+			testFunc: func(t *testing.T) {
+				mutateFileOnNode(t, c.ClientSet, c.node, configDriftFilename, "not-the-data")
+				assertNodeAndMCPIsDegraded(t, c.ClientSet, c.node, c.mcp, configDriftFilename)
+
+				node, err := c.ClientSet.CoreV1Interface.Nodes().Get(context.TODO(), c.node.Name, metav1.GetOptions{})
+				require.Nil(t, err)
+
+				firstReason := node.Annotations[constants.MachineConfigDaemonReasonAnnotationKey]
+
+				mutateFileOnNode(t, c.ClientSet, c.node, configDriftFilename, configDriftFileContents)
+				assertNodeAndMCPIsRecovered(t, c.ClientSet, c.node, c.mcp)
+
+				mutateFileOnNode(t, c.ClientSet, c.node, configDriftFilenameTwo, "incorrect data 2")
+				assertNodeAndMCPIsDegraded(t, c.ClientSet, c.node, c.mcp, configDriftFilenameTwo)
+
+				mutateFileOnNode(t, c.ClientSet, c.node, configDriftFilename, "not-the-data")
+				assertNodeAndMCPIsDegraded(t, c.ClientSet, c.node, c.mcp, configDriftFilename)
+
+				node, err = c.ClientSet.CoreV1Interface.Nodes().Get(context.TODO(), c.node.Name, metav1.GetOptions{})
+				require.Nil(t, err)
+
+				assert.Equal(t, node.Annotations[constants.MachineConfigDaemonReasonAnnotationKey], firstReason)
+
+				assertPoolReachesState(t, c.ClientSet, c.mcp, func(m mcfgv1.MachineConfigPool) bool {
```

Comment:
Since this looks like a duplicate of the closure used in `assertNodeAndMCPIsDegraded`, lets hoist this (and the other closure contents) into a separate function called, e.g., `func isPoolDegraded(m mcfgv1.MachineConfigPool) bool`. Then this line and the other can become something like:

`assertPoolReachesState(t, c.ClientSet, c.mcp, isPoolDegraded)`